service/stub_module.go

```diff
@@ -33,14 +33,8 @@ type StubModule struct {
 var _ Module = &StubModule{}
 
 // NewStubModule generates a Module for use in testing
-func NewStubModule() *StubModule {
-	return &StubModule{}
-}
-
-// Initialize fakes an init call on the module
-func (s *StubModule) Initialize(host Host) error {
-	s.Host = host
-	return s.InitError
+func NewStubModule(host Host) *StubModule {
+	return &StubModule{Host: host}
```

Comment:
To answer the question, today the host is always initialized by the time we add these modules. We don't have use cases today for where it will not be. When it comes up, we can revisit and see if Initialize is the solution or we should handle it some other way.